code/datums/status_effects/wound_effects.dm

```diff
@@ -145,14 +145,23 @@
 // bones
 /datum/status_effect/wound/blunt
 
-/datum/status_effect/wound/blunt/interact_speed_modifier()
+/datum/status_effect/wound/blunt/on_apply()
+	. = ..()
+	RegisterSignal(owner, COMSIG_MOB_SWAP_HANDS, .proc/on_swap_hands)
+	on_swap_hands()
+
+/datum/status_effect/wound/blunt/on_remove()
+	. = ..()
+	UnregisterSignal(owner, COMSIG_MOB_SWAP_HANDS)
 	var/mob/living/carbon/C = owner
+	C.remove_actionspeed_modifier(/datum/actionspeed_modifier/blunt_wound)
 
+/datum/status_effect/wound/blunt/proc/on_swap_hands()
+	var/mob/living/carbon/C = owner
```

Comment:
Better variable names, please.

Reply:
tell ryll :^(

but yeah sure ill fix it